packages/calling-component-bindings/src/callControlSelectors.ts

```diff
@@ -109,12 +115,23 @@ export type ScreenShareButtonSelector = (
  * @public
  */
 export const screenShareButtonSelector: ScreenShareButtonSelector = reselect.createSelector(
-  [getIsScreenSharingOn, /* @conditional-compile-remove(PSTN-calls) */ getCallState],
-  (isScreenSharingOn, /* @conditional-compile-remove(PSTN-calls) */ callState) => {
+  [
+    getIsScreenSharingOn,
+    /* @conditional-compile-remove(PSTN-calls) */ getCallState,
+    /* @conditional-compile-remove(rooms) */ getRole
+  ],
+  (
+    isScreenSharingOn,
+    /* @conditional-compile-remove(PSTN-calls) */ callState,
+    /* @conditional-compile-remove(rooms) */ role
+  ) => {
     return {
       checked: isScreenSharingOn,
       /* @conditional-compile-remove(PSTN-calls) */
-      disabled: callState === 'InLobby' ? true : callState === 'Connecting' ?? false
+      disabled:
+        (callState === 'InLobby' ? true : callState === 'Connecting' ?? false) ||
+        /* @conditional-compile-remove(rooms) */ role === 'Consumer' ||
+        /* @conditional-compile-remove(rooms) */ role === 'Attendee'
```

Comment:
I will override the changes in this file in [PR](https://github.com/Azure/communication-ui-library/pull/3374/files#diff-5fc9f58576485df70f1f052afe532582e08f93d9ff12bc0f35df0111c0475abf) but this is a good intermediate.

Reply:
I think we can still do the mapping through roles as a condition as well. since rooms might not respect the capabilities the same way as interop